services/s3/src/test/java/software/amazon/awssdk/services/s3/S3UtilitiesTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import org.junit.AfterClass;
+import org.junit.Test;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.model.GetUrlRequest;
+
+public class S3UtilitiesTest {
+
+    private static final URI US_EAST_1_URI = URI.create("https://s3.amazonaws.com");
+
+    private static final S3Configuration ACCELERATE_AND_DUALSTACK_ENABLED = S3Configuration.builder()
+                                                                                     .accelerateModeEnabled(true)
+                                                                                     .dualstackEnabled(true)
+                                                                                     .checksumValidationEnabled(true)
+                                                                                     .build();
+    private static final S3Configuration PATH_STYLE_CONFIG = S3Configuration.builder()
+                                                                   .pathStyleAccessEnabled(true)
+                                                                   .build();
+
+    private static final S3Client defaultClient = S3Client.create();
+    private static final S3Utilities defaultUtilities = defaultClient.utilities();
```

Comment:
Can we add the assertions in the same tests here to cover utilities for async clients?

Reply:
Added a test using async client